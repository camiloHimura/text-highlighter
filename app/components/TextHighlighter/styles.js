import styled from 'styled-components'


export const Container = styled.div`
  width: 100%;
  padding: 6px;
  height: 40vh;
  border: 1px solid black;

& > textarea, & > div{
  width: 100%;
  height: 100%;
  resize: none;
  border: none;
  outline: none;
  font-size: 16px;
  line-height: 16px;
  font-family: Times;
}

&.red textarea::selection{
  background-color: red;
}
&.yellow textarea::selection{
  background-color: yellow;
}
&.green textarea::selection{
  background-color: green;
}
`;
