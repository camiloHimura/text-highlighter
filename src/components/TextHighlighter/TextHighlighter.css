.textHighlighter{
    width: 100%;
    padding: 6px;
    height: 40vh;
    border: 1px solid black;
}

.textHighlighter > textarea, .textHighlighter > div{
    width: 100%;
    height: 100%;
    resize: none;
    border: none;
    outline: none;
    font-size: 16px;
    line-height: 16px;
    font-family: Times;
}

.textHighlighter.red textarea::selection{
    background-color: red;
}
.textHighlighter.yellow textarea::selection{
    background-color: yellow;
}
.textHighlighter.green textarea::selection{
    background-color: green;
}

