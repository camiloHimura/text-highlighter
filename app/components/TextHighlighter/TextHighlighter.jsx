import React, {useState, useRef} from 'react';
import {connect} from 'react-redux';
import PropTypes from "prop-types";
import {Container} from "./styles.js"

import {addTextAction} from "../../state/actions";

const mapStateToProps = state => {
  return {
        selectHighLighter: state.colors.selectHighLighter,
      }
}

const mapDispachToProps = dispatch => {
  return {
    addText: (text, color) => dispatch(addTextAction({text, color})),
  }
}

export function TextHighlighter(props){
  const {optionSelected = "", selectHighLighter, addText} = props;
  const inputTextarea = useRef();

  function selectedText(event){
    const {target} = event;
    const textSlected = target.value.substring(target.selectionStart, target.selectionEnd);
    addText(textSlected, selectHighLighter);
  }

  return  <Container className={optionSelected}>
            <textarea autoFocus={true} ref={inputTextarea} onSelect={selectedText}>
            </textarea>
          </Container>
}

TextHighlighter.propTypes = {
  optionSelected: PropTypes.string,
  selectHighLighter: PropTypes.string,
  addText: PropTypes.func,
}

export default connect(mapStateToProps, mapDispachToProps)(TextHighlighter);
