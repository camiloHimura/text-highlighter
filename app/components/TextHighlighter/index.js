import TextHighlighter from "./TextHighlighter.jsx";
export default TextHighlighter;