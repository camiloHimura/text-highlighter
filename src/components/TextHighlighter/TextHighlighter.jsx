import React, {useState, useRef} from 'react';
import {connect} from 'react-redux';
import "./TextHighlighter.css"

import {addTextAction} from "../../state/actions";

const mapStateToProps = state => {
    return {
            selectHighLighter: state.colors.selectHighLighter,
        }
}

const mapDispachToProps = dispatch => {
    return {
        addText: (text, color) => dispatch(addTextAction({text, color})),
    }
}

function TextHighlighter(props){
    const {optionSelected = "", selectHighLighter, addText} = props;

    const inputTextarea = useRef();

    function selectedText(event){
        const {target} = event;
        const textSlected = target.value.substring(target.selectionStart, target.selectionEnd);
        addText(textSlected, selectHighLighter);
    }

    const className = `textHighlighter ${optionSelected}`
    return  <div className={className}>
                <textarea autoFocus={true} ref={inputTextarea} onSelect={selectedText}>
                </textarea>
            </div>
}

export default connect(mapStateToProps, mapDispachToProps)(TextHighlighter) ;