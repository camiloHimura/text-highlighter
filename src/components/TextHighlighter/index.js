import TextHighlighter from "./TextHighlighter";
export default TextHighlighter;