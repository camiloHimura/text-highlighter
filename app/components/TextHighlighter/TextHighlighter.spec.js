import React from 'react';
import { TextHighlighter } from './TextHighlighter';
import { shallow } from 'enzyme'


describe('Colors component', () => {
  let Component;
  let props;

  beforeEach(() => {
    props = {
      addText: jest.fn(),
      optionSelected: 'red',
      selectHighLighter: 'red',
    }
    Component = shallow(<TextHighlighter {...props}/>);
  })
  it('take snapshot', () => {
    expect(Component).toMatchSnapshot();
  });

  it('highlighting text', () => {
    const data = {value: 'this is a test', selectionStart: 0, selectionEnd:3};
    Component.find('textarea').simulate('select', { target: data});
    const textSlice = data.value.slice(data.selectionStart, data.selectionEnd);
    expect(props.addText).toHaveBeenCalledWith(textSlice, props.selectHighLighter);
  });
})
