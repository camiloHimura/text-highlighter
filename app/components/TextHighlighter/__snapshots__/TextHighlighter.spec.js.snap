// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component take snapshot 1`] = `
<styled.div
  className="red"
>
  <textarea
    autoFocus={true}
    onSelect={[Function]}
  />
</styled.div>
`;
